tests/store/advocates.spec.ts

```diff
@@ -0,0 +1,294 @@
+import advocatesModule from '~/store/modules/advocates'
+
+/**
+ * ACTIONS
+ * -----------------------------------------------------------------------------
+ */
+
+describe('fetchAdvocates', () => {
+  it('fetches the list of advocates', async () => {
```

Comment:
This test is actually also checking the following assertion in line 17: `expect.arrayContaining([expect.anything()])`.

This would only pass if the advocates list has been fetched correctly.

Reply:
By "fetches the list of advocates", I understand that you can check if the list fetched is the proper one. For instance:

```
 await fetchAdvocates({ commit })
 expect(state.advocates).toEqual([{advocate1, advocate2}])
```

With your code, you check that "something" is fetched, and IMO, that doesn't test that you get the correct advocates.

Maybe you should change the description instead of the code

Reply:
From personal preference I think I'd go with something like _fetches the **correct** list of advocates_ if were to test for the correctness of the advocates list.

To fully test if the action if fetching a list of advocates instead of a different kind of list, I think I would check for the shape of the array elements. But I believe that this would result in higher maintenance for a simple test, and thus defeat the helpfulness of it for our specific project.

I agree that the description of the test as you point out can be misleading though, so I just changed it a little to try and reflect more what the test actually checks. 🙂